gpt_index/response/schema.py

```diff
@@ -19,11 +19,20 @@ class SourceNode(DataClassJsonMixin):
 
     source_text: str
     doc_id: Optional[str]
+    extra_info: Optional[Dict[str, Any]] = None
```

Comment:
conceptually extra_info is carried over from the original Document right? if that's the case do we need to store on SourceNode vs. just using doc_id to look up the original document?

Reply:
i could see an argument for this if extra_info has more stuff (say from the Node) than just what's carried over from the original Document

Reply:
Yeah technically you can always look up the doc_id but I found it to be inconvenient because you'd have to perform the lookup for every SourceNode before returning the result to the Front End, since there's no concept of documents on the FE. Also, it's possible that in the future, a Node's extra_info may diverge from the Document's extra_info (like in your earlier proposal). To me, it feels natural to have the important attributes of the Node included in the Source Node.

IMO dream scenario is that you get a response from `index.query`, forward that to the FE, and you have everything you need to make a complete UX (including metadata and similarity scores).

Reply:
Hm ok! we can keep if it's easier for this given use case